doc/design/codegen.md

```diff
@@ -0,0 +1,95 @@
+# SQLFlow Code Generator
+
+SQLFlow is a compiler that takes a SQL program and generates Argo workflow, which is a `.YAML` file,
+The Argo controller running on Kubernetes is the executor. This is a design doc on how to implement
+the back-end of the SQLFlow compiler.
+
+## The High-level Design of the Code Generator
+
+As mentioned above, SQLFlow compiler generates a `.YAML`, the piece code is as the following:
+
+``` yaml
+steps:
+    name: step-1
+    command: ["python", "-c"]
```

Comment:
Is the command always be `python -c`?
For TO RUN statement, if we execute a binary instead of a python script, the command is the executable name directly.

Reply:
No, actually each code generator can define the execution command as the description of `ExeuctionCtx` at L57